faunadb/client.go

```diff
@@ -56,6 +62,22 @@ func DisableTxnTimePassthrough() ClientConfig {
 	return func(cli *FaunaClient) { cli.isTxnTimeEnabled = false }
 }
 
+//QueryConfig is the base type for query specific configuration parameters.
+type QueryConfig func(FaunaClient, *faunaRequest)
```

Comment:
I don't think we need `FaunaClient` here. It is, after all, concerned itself with request specific configuration. Any other client configuration should be done by the client constructor at this point.